esrally/mechanic/launcher.py

```diff
@@ -327,13 +321,14 @@ def _set_env(self, env, k, v, separator=' '):
                 env[k] = v + separator + env[k]
 
     @staticmethod
-    def _start_process(binary_path):
+    def _start_process(binary_path, env):
         if os.geteuid() == 0:
             raise exceptions.LaunchError("Cannot launch Elasticsearch as root. Please run Rally as a non-root user.")
         os.chdir(binary_path)
         cmd = ["bin/elasticsearch"]
         cmd.extend(["-d", "-p", "pid"])
         proc = subprocess.Popen(cmd,
+                                env,
```

Comment:
You need to use a keyword-arg here (i.e. `env=env`) otherwise it treats this argument as positional and at this position it expects the `bufsize` argument. You can check this e.g. by running:

```
esrally --test-mode --distribution-version=1.7.6 --on-error=abort --challenge=append-no-conflicts-index-only
```

This fails with:

```
[ERROR] Cannot race. bufsize must be an integer
```